pkg/operator/controller/nodeobservability/controller.go

```diff
@@ -313,7 +322,34 @@ func (r *NodeObservabilityReconciler) ensureNodeObservabilityDeleted(ctx context
 
 // machineConfigChangeRequested returns true, when a given NodeObservabilityType needs
 // machine config change. Only CrioKubeletNodeObservabilityType requires a MC change, false otherwise
-func machineConfigChangeRequested(nodeObs *operatorv1alpha2.NodeObservability) bool {
+func (r *NodeObservabilityReconciler) machineConfigChangeRequested(ctx context.Context, nodeObs *operatorv1alpha2.NodeObservability) bool {
+	// avoid the creation of a new NOMC if the CRIO profiling is already enabled in the rendered MC
+	mcpName := types.NamespacedName{Name: workerMCP}
+	mcp := &mcv1.MachineConfigPool{}
+	err := r.ClusterWideClient.Get(ctx, mcpName, mcp)
+	if err != nil {
+		r.Log.Error(err, "failed to get the machineconfigpool instance "+mcpName.Name)
+		return nodeObs.Spec.Type == operatorv1alpha2.CrioKubeletNodeObservabilityType
+	}
+	// getting the rendered mc name from the obtained machine config pool resource
+	mcName := types.NamespacedName{Name: mcp.Spec.Configuration.Name}
+	renderedMC := &mcv1.MachineConfig{}
+	err = r.ClusterWideClient.Get(ctx, mcName, renderedMC)
+	if err != nil {
+		r.Log.Error(err, "failed to get the machineconfig instance "+mcName.Name)
+		return nodeObs.Spec.Type == operatorv1alpha2.CrioKubeletNodeObservabilityType
+	}
+	bData, err := yaml.Marshal(renderedMC.Spec.Config)
```

Comment:
nit: `machineconfig` controller uses `json` to marshal/unmarshal, we could keep both in sync.

Reply:
I figured out during my testing that this won't work as the Config has a nested data.
Anyway, there is a Marshal() method for the `Spec.Config` that's working (verified)